reference/5.1/Microsoft.PowerShell.Core/About/about_PowerShell_exe.md

```diff
@@ -157,6 +157,13 @@ It is not possible to pass an explicit boolean value for a switch parameter
 when running a script in this way. This limitation was removed in PowerShell 6
 (`pwsh.exe`).
 
+> [!NOTE]
+> The **File** parameter cannot support scripts using a parameter that expects
+> an array of argument values. This, unfortunately, is a limitation of how a
+> native command gets argument values. When you call a native executable (such
+> as `powershell` or `pwsh`), it does not know what to do with an array, so
+> it's passed as a string.
```

Comment:
Is the array _passed_ as a string or _parsed by the executable_ as a string? If I type

```powershell
pwsh -File .\Get-Script.ps1 -Foo a, b, c
```

The value of **Foo** in the script is `'a, b, c'`, correct? But what if I pass three non-string objects? Are those cast to a string and passed, or what happens? Can I pass a variable containing the array?

```powershell
$StringArray = @('a', 'b', 'c')
$HashArray = @(@{ Name = 'Bar' }, @{ Name = 'Baz' })
pwsh -File .\Get-Script.ps1 -Foo @('a', 'b', 'c')
pwsh -File .\Get-Script.ps1 -Foo $StringArray
pwsh -File .\Get-Script.ps1 -Foo @{ Name = 'Bar'}, @{ Name = 'Baz'}
pwsh -File .\Get-Script.ps1 -Foo $HashArray
```

Reply:
Oh, duh. I was thinking of going from PowerShell to PowerShell because I haven't had enough coffee. There's no good reason to do that (or document it) since if you're already in PowerShell... you'd just run the script. Sorry for the tangent!